### DISCUSSION TITLE
Correct way to install zeek plugn - ICSNPP-BACnet

### DISCUSSION THREAD

**Discussion by mintos5** [Security-Onion-Solutions/securityonion#8959] [Unsupported Versions] [answered]
Hi all,

i am using custom Zeek container based on the newest SO Dockerfile:
https://github.com/Security-Onion-Solutions/securityonion-image/blob/master/so-zeek/Dockerfile

This new Dockerfile contains command to install **icsnpp-bacnet** (added just recently - october 4) but I had problems getting the output logs from this project: bacnet.log, bacnet_property.log or bacnet_discovery.log

I edited needed pillar config, so local.zeek has line: **@load icsnpp-bacnet**

But script is not working properly:
```
logger scripts failed.
error in /opt/zeek/share/zeek/site/icsnpp-bacnet/./main.zeek, line 120: unknown identifier bvlc_functions, at or near "bvlc_functions"

manager scripts failed.
error in /opt/zeek/share/zeek/site/icsnpp-bacnet/./main.zeek, line 120: unknown identifier bvlc_functions, at or near "bvlc_functions"

proxy scripts failed.
error in /opt/zeek/share/zeek/site/icsnpp-bacnet/./main.zeek, line 120: unknown identifier bvlc_functions, at or near "bvlc_functions"

worker-1-1 scripts failed.
error in /opt/zeek/share/zeek/site/icsnpp-bacnet/./main.zeek, line 120: unknown identifier bvlc_functions, at or near "bvlc_functions"
```

I think the problem is because of missing **consts.zeek** file, that is defining bvlc_functions.
https://github.com/cisagov/icsnpp-bacnet/blob/main/scripts/consts.zeek

I fixed the problem by commenting rm commands in DockerFile:
```
    /opt/zeek/bin/zkg install --force zeek-spicy-stun && \
#    rm -rf /opt/zeek/var/lib/zkg/testing && \
#    rm -rf /opt/zeek/var/lib/zkg/clones && \
#    rm -rf /opt/zeek/var/lib/zkg/scratch
RUN groupadd --gid 937 zeek  && \
```
After this fix i am getting needed logs. I verified that Zeek is really loading some files from removed filesystem:
```
{"name":"/opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/scripts/__preload__.zeek"}
{"name":"  /opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/scripts/consts.zeek"}
{"name":"/opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/lib/bif/__load__.zeek"}
{"name":"  /opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/lib/bif/events.bif.zeek"}
{"name":"/opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/scripts/__load__.zeek"}
...
{"name":"  /nsm/zeek/spool/installed-scripts-do-not-touch/site/icsnpp-bacnet/__load__.zeek"}
{"name":"    /nsm/zeek/spool/installed-scripts-do-not-touch/site/icsnpp-bacnet/main.zeek"}
```
Path **/opt/zeek/lib64/zeek/plugins/packages/icsnpp-bacnet/scripts/** is only symlink to **/opt/zeek/var/lib/zkg/clones/package/icsnpp-bacnet/scripts**, that is removed with rm in DockerFile.

So is there a better way to enable **icsnpp-bacnet** plugin/script than commenting rm commands in DockerFile?

**Comment by dougburks** [comment, marked as answer]
The recent Zeek plugins are a work in progress and we have more work to do.
